iina/AppDelegate+Scripting.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  AppDelegate+Scripting.swift
+//  iina
+//
+//  Created by Nate Weaver on 2020-03-02.
+//  Copyright © 2020 lhc. All rights reserved.
+//
+
+import AppKit
+
+extension AppDelegate {
+
+  func application(_ sender: NSApplication, delegateHandlesKey key: String) -> Bool {
+    return Set(["orderedPlayerWindows", "orderedPlayers"]).contains(key)
+  }
+
+  @objc var orderedPlayers: [PlayerCore] {
+    let windows = NSApp.orderedWindows
+    var players = [PlayerCore]()
+
+    for window in windows {
+      if window.isVisible, let controller = window.delegate as? PlayerWindowController {
+        players.append(controller.player)
+      }
+    }
+
+    return players
```

Comment:
This could all probably be done functionally, e.g.
```
Array(NSApp.orderedWindows.lazy.filter(\.isVisible).flatMap { $0.delegate as? PlayerWindowController}.map(\.player))
```